packages/flame/lib/src/effects/move_to_effect.dart

```diff
@@ -40,7 +40,7 @@ class MoveToEffect extends MoveEffect {
   @override
   void apply(double progress) {
     final dProgress = progress - previousProgress;
-    target.position += _offset * dProgress;
+    target.position += _offset * dProgress.abs();
```

Comment:
Hmm, is this really correct?

Reply:
After thinking about it some more yesterday I think the issue wasn't the reflected vector itself, but rather that the direction vector was incorrect when the camera has a viewport-aware bounds behavior. This causes it to never reach its destination and so the reflective vector is incorrect. That's why taking the absolute value was ok after I added the step to recalculate the vector.

The camera won't be the only component that may never complete its effect with regards to `MoveToEffect` so this matter should be resolved in the PR imo.

Reply:
But this will potentially overshoot the destination right? And also it won't be matching the time anymore?
If you are at 0.99 progress and the delta progress is -0.02 when the effect is going forward it should have moved 0.03 "steps", since it'll start going back again, but with this it will simply move to 1.01.

Reply:
I'm not sure I'm understanding your question precisely, but it seems that you've noticed the possibility of under-fitting the expected step amounts, time-wise. Yes this does happen with the viewport-aware behavior as described (the camera _cannot_ exceed the bounds, and therefore never arrives to its destination). ATM with the changes I've added, I have not seen any noticeable problems.

Reply:
Simply put; I don't think it will always be at the expected place at each time step when `abs` is used, that was what I tried to describe in the last comment, the delta doesn't become correct. If it overshoots its destination it should start going back, in the cases where it is alternating.

Reply:
We make explicit use of `onStart` so that we are recalculating the vector from the equation `v = to - from` even when the pattern has reversed. This means our progress values shouldn't be negative otherwise it will reverse the vector `v`, which already has correct directionality in its scalar's sign from the explicit call to `onStart`.

Reply:
It does sound like it makes sense but I'm on my phone, but if that is what is happening, then how does it then make sense to make it `abs`?

Reply:
`progress` will be decrementing `dt` units from `1.0` during the reverse phase of the alternating pattern, while `previousProgress` retains the `progress+dt` value form before. Therefore it would be negative, which we don't want to use. It's better to recalculate the starting position and use positive scalars instead.

Reply:
But it'll surely overshoot the end value then?

Reply:
Hi. No it never does for this effect or any as far as I can tell.

Reply:
In the case that `dProgress` becomes negative it is supposed to go backwards. Say that the previous progress is 0.9 and it it supposed to move backwards to 0.8, then `dProgress` should be `-0.1`, not `0.1`.
Try running only the the `MoveToEffect` with `alternate: true`.